### PR TITLE
updating to nx 16 starting point

### PR DESCRIPTION
**Pull request by barbados-clemens**
Hi all!

As Nx preps for the v16 release, we like to sanity check things against community repos. In that process we end up doing a chuck of work of the migration and I wanted to present that as a base for y'all to finish the migration. 

Since I don't fully understand the repo and the domain knowledge that has grown with it, this PR doesn't fully _work_ but is a great base for the maintainers to be able to push the last bit. 

The biggest thing I came across is the change Nx v16 made to move a lot of utilities out of workspace package into the js/devkit package. Most of these map 1 to 1 over or are actually simpler as being apart of the API surface of the various packages. i.e. runTaskInSerial is now exported from `@nx/devkit` instead of from a deep import from `@nx/workspace`. A lot of AST utils are now apart of the public API of `@nx/js`.

> Side note packages have been rescoped to the `@nx` npm scope now, the `@nrwl` scope version of the packages are also published along side for now as just reexports of the `@nx` scope packages and will eventually stop being published.

This PR should allow all projects to be able to be built, `nx run-many -t build`
but tests/lint aren't fully working mainly because of older migration files built around the angular schematics devkit and incompatibility with Nx utils since some have changed/removed.

From what I've seen, most of the migrations would probably be deleted since they are very old, but this depends on how many versions at a time is supported. i.e. Nx only maintains the last 2 major versions of migrations for each plugin. This is because older plugins become maintenance over time trying to stay 'up to date' with api surfaces of a plugin, while support 'old' versions of the plugin. 

The main issue I ran into are the tests that check the versions were updated for a given migration. I believe at this point you shouldn't have to manually do this anymore as the `migration.json#packageJsonUpdates` field will get those versions auto applied. [here is an example of how Nx plugins manage it](https://github.com/nrwl/nx/blob/master/packages/js/migrations.json#L46-L72).

There were also a couple changes to some underlying Nx plugins that needed to be updated, mainly Jest and Cypress now are async functions.

Hopefully this is helpful for y'all and thank you for making awesome plugins for the Nx community 😊

### CODE REVIEW

File: packages/ionic-angular/package.json [hunk 1]
```diff
@@ -26,10 +26,10 @@
...
   "dependencies": {
     "@nxext/capacitor": "^15.7.0",
-    "@nrwl/devkit": "15.7.0",
-    "@phenomnomnominal/tsquery": "4.1.1"
+    "@phenomnomnominal/tsquery": "4.1.1",
+    "@nx/devkit": "16.0.0-beta.7"
   },
   "peerDependencies": {
-    "@nrwl/angular": "^15.7.0"
+    "@nx/angular": "16.0.0-beta.7"
```
Comment (barbados-clemens) [inline packages/ionic-angular/package.json:33]: unsure what the deps policy is here, but the Nx migration pinned the deps. might need this switched out to `^<version>`

File: packages/solid/src/generators/component/lib/add-exports-to-barrel.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import {
   readProjectConfiguration,
   StringChange,
   Tree,
-} from '@nrwl/devkit';
+} from '@nx/devkit';
 import { findNodes } from 'nx/src/utils/typescript';
```
Comment (barbados-clemens) [inline packages/solid/src/generators/component/lib/add-exports-to-barrel.ts:13]: I believe this is now in @nx/js public api now.

File: packages/stencil/src/executors/build/executor.ts [hunk 3]
```diff
@@ -5,14 +5,14 @@ import {
   createStencilProcess,
   initializeStencilConfig,
 } from '../stencil-runtime';
-import { createProjectGraphAsync } from '@nrwl/workspace/src/core/project-graph';
+import { createProjectGraphAsync } from '@nx/workspace/src/core/project-graph';
```
Comment (barbados-clemens) [inline packages/stencil/src/executors/build/executor.ts:8]: should import from devkit

**Comment by DominikPieper** [comment]
Oh wow, thank you, really much @barbados-clemens. That's more than great!

**Comment by barbados-clemens** [comment]
Let me know if the changes Nx made aren't clear or are breaking anything unexpectedly so we can get it patched up! happy to help :)

File: packages/ionic-angular/src/generators/application/lib/add-angular.ts [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 import { readNxVersion } from '../../../utils/utils';
 
 export async function addAngular(
   host: Tree,
   options: ApplicationGeneratorSchema
 ) {
-  await ensurePackage(host, '@nrwl/angular', readNxVersion(host));
-  const { applicationGenerator } = await import('@nrwl/angular/generators');
+  ensurePackage('@nx/angular', readNxVersion(host));
+  const { applicationGenerator } = await import('@nx/angular/generators');
```
[the remark this answers is not in the document]
Comment (DominikPieper) [reply packages/ionic-angular/src/generators/application/lib/add-angular.ts:11]: That's great. I'll refactor everything :-D

**Pull request merged**